discovery_engine_core/bindings/src/types/primitives.rs

```diff
@@ -14,13 +14,42 @@
 
 //! Modules containing FFI glue for handling primitives (expect `str`/`slice`).
 
+use std::convert::TryInto;
+
 use super::{
     boxed,
     option::get_option_some,
     slice::{alloc_uninitialized_slice, boxed_slice_from_raw_parts},
     vec::{alloc_vec, get_vec_buffer, get_vec_len},
 };
 
+/// Type to represent "length" in FFI.
+///
+/// Be aware that values returned are clamped
+/// to 2**32, e.g. for `Vec` the extern interface will
```

Comment:
clamping would require a lower bound as well
```suggestion
/// Be aware that values returned are bounded
/// by 2**32, e.g. for `Vec` the extern interface will
```